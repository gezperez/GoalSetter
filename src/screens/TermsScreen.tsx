import React from 'react'
import { Dimensions, Image, ImageBackground, ImageStyle, StyleSheet, Text, TextStyle } from 'react-native'

import { Container } from '~/components'
import I18n from '~/i18n'
import { Images } from '~/utils'
import Typography from '~/utils/Typographys'

const WIDTH = Dimensions.get('window').width

const TermsScreen = () => {
  return (
    <Container>
      <ImageBackground source={Images.MONEY_BACKGROUND} style={styles.imageBackground}>
        <Image source={Images.FLYING_MONEY} />
      </ImageBackground>
      <Text style={styles.text}>{I18n.t('terms')}</Text>
    </Container>
  )
}

type Styles = {
  imageBackground: ImageStyle
  text: TextStyle
}

const styles = StyleSheet.create<Styles>({
  imageBackground: {
    width: WIDTH * 0.7,
    height: 45,
    alignSelf: 'center',
    justifyContent: 'center',
    alignItems: 'center',
    marginVertical: 32,
  },
  text: {
    ...Typography.text,
    margin: 32,
  },
})

export default TermsScreen
