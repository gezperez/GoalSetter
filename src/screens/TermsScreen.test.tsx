import React from 'react'
import { SafeAreaProvider } from 'react-native-safe-area-context'
import { render } from '@testing-library/react-native'

import I18nJs from '~/__mocks__/react-native-i18n'
import TermsScreen from './TermsScreen'

// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
// @ts-ignore
global.React = React

describe('TermsScreen', () => {
  afterEach(jest.resetAllMocks)

  it('should render with Button component', () => {
    const { toJSON, getByText } = render(
      <SafeAreaProvider>
        <TermsScreen />
      </SafeAreaProvider>
    )

    const text = getByText(I18nJs.t('terms'))
    expect(text).toBeDefined()
    expect(toJSON()).toMatchSnapshot()
  })
})
