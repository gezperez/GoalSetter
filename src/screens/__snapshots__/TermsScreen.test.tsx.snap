// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsScreen should render with Button component 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Array [
        undefined,
        Object {
          "flexGrow": 1,
        },
      ]
    }
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "height": 45,
            "justifyContent": "center",
            "marginVertical": 32,
            "width": 525,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/money-background.png",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 45,
                "width": 525,
              },
              undefined,
            ]
          }
        />
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/images/flying-money.png",
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "Montserrat-Light",
            "fontSize": 14,
            "lineHeight": 16,
            "margin": 32,
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. 


  1.  Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.n. .
  
  2. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
  
  3. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. 
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
